File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/options/SetVolumeOptions.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.youtubeplayer.YoutubePlayerHelper;
import io.capawesome.capacitorjs.plugins.youtubeplayer.classes.CustomExceptions;

public class SetVolumeOptions {

    @NonNull
    private final String id;

    private final int volume;

    public SetVolumeOptions(@NonNull PluginCall call) throws Exception {
        this.id = YoutubePlayerHelper.getPlayerIdFromCall(call);
        this.volume = SetVolumeOptions.getVolumeFromCall(call);
    }

    @NonNull
    public String getId() {
        return id;
    }

    public int getVolume() {
        return volume;
    }

    private static int getVolumeFromCall(@NonNull PluginCall call) throws Exception {
        Integer volume = call.getInt("volume");
        if (volume == null || volume < 0 || volume > 100) {
            throw CustomExceptions.VOLUME_INVALID;
        }
        return volume;
    }
}

File: packages/youtube-player/ios/Plugin/Classes/Options/SetVolumeOptions.swift
import Capacitor
import Foundation

@objc public class SetVolumeOptions: NSObject {
    let id: String
    let volume: Int

    init(_ call: CAPPluginCall) throws {
        self.id = try YoutubePlayerHelper.getPlayerIdFromCall(call)
        self.volume = try SetVolumeOptions.getVolumeFromCall(call)
    }

    private static func getVolumeFromCall(_ call: CAPPluginCall) throws -> Int {
        guard let volume = call.getInt("volume"), volume >= 0, volume <= 100 else {
            throw CustomError.volumeInvalid
        }
        return volume
    }
}
